parastell/radial_distance_utils.py

```diff
@@ -1,154 +1,368 @@
 
-def get_start_index(filament):
-    """
-    Find the index at which the filament crosses the xy plane on the OB
-    side of the filament
+
+def reorder_filament(coil):
+    """Reorders the filament coordinates of a MagnetCoil class object such that
+    they begin near the outboard midplane, and initially progress positively in
+    the z-direction.
 
     Arguments:
-        filament (list of list of float): List of points defining the filament
+        coil (object): MagnetCoil class object.
 
     Returns:
-        crossing_index (int): index at which the filament crosses the xy plane
-            on the OB side of the filament.
-    """
-    crossing_index = None
-    max_crossing_radius = 0
-    for index, (point, next_point) in enumerate(
-        zip(filament[0:-1], filament[1:])
-    ):
-        if point[2] / next_point[2] < 0:
-            crossing_radius = calc_z_radius(point)
-            if max_crossing_radius < crossing_radius:
-                crossing_index = index
-                max_crossing_radius = crossing_radius
-    return crossing_index
-
-
-def sort_filaments_toroidally(filaments):
+        reordered_coords (2-D iterable of float): reordered list of Cartesian
+            coordinates defining a MagnetCoil filament.
     """
-    Reorder filaments in order of increasing toroidal angle
+    # Start the filament at the outboard midplane
+    outboard_index = get_start_index(coil)
+    reordered_coords = np.concatenate(
+        [coil.coords[outboard_index:], coil.coords[1:outboard_index]]
+    )
+    # Ensure filament is a closed loop
+    if outboard_index != 0:
+        reordered_coords = np.concatenate(
+            [reordered_coords, [reordered_coords[0]]]
+        )
+    # Ensure points initially progress in positive z-direction
+    if reordered_coords[0, 2] > reordered_coords[1, 2]:
+        reordered_coords = np.flip(reordered_coords, axis=0)
+    coil.coords = reordered_coords
+
+
+def sort_coils_toroidally(magnet_coils):
+    """Reorders list of coils by toroidal angle on range [-pi, pi] (coils
+    ordered in MagnetSet class by toroidal angle on range [0, 2*pi]).
 
     Arguments:
-        filaments (list of list of list of float): List of filaments, which are
-            lists of points defining each filament.
+        magnet_coils (list of object): list of MagnetCoil class objects.
+
     Returns:
-        filaments (list of list of list of float): filaments in order of
-            increasing toroidal angle.
+        magnet_coils (list of object): sorted list of MagnetCoil class objects.
     """
-    com_list = np.zeros((len(filaments), 3))
+    com_list = np.array([coil.center_of_mass for coil in magnet_coils])
+    com_toroidal_angles = np.arctan2(com_list[:, 1], com_list[:, 0])
 
-    for idx, fil in enumerate(filaments):
-        com_list[idx] = np.average(fil, axis=0)
+    magnet_coils = np.array(
+        [x for _, x in sorted(zip(com_toroidal_angles, magnet_coils))]
+    )
```

Comment:
If you define `MagnetCoil.com_toroidal_angle()`
```suggestion
    magnet_coils =  sorted(magnet_coils, key = lambda x: x.com_toroidal_angle)
```